evm/src/modules/TokenGateway.sol

```diff
@@ -48,51 +77,109 @@ contract TokenGateway is IIsmpModule {
     // set the ismp host address
     function setIsmpHost(address _host) public onlyAdmin {
         host = _host;
-        admin = address(0);
+    }
+
+    function setTokenIdentifierDetails(bytes32 _tokenId, address _erc20, address _erc6160) external onlyAdmin {
+        _erc6160s[_tokenId] = _erc6160;
+        _erc20s[_tokenId] = _erc20;
+    }
```

Comment:
Should be done individually

Reply:
unresolved

Reply:
useful for batch actions, saves both runtime and deployment gas costs

Reply:
erc6160s can exist without an erc20 counterpart, lets split them out.

Reply:
setting it to address(0) still maintains it, but okay, done